server/api/src/rsGeneralAdmin.cpp

```diff
@@ -52,6 +53,63 @@ namespace
         return std::find_if(b, e, [](unsigned char _ch) { return ::isspace(_ch); }) != e;
     } // contains_whitespace
 
+     auto throw_if_inappropriate_rodsadmin_downgrade(
+        RsComm& rsComm,
+        const std::string_view& _new_value,
+        const std::string_view& _user_name
+    ) -> void {
+            std::set<std::string> server_administrators_in_zone;
+
+            BytesBuf* bbuf = nullptr;
+            irods::at_scope_exit free_buf { [&bbuf] { std::free(bbuf); }  };
+
+            if (const auto ec = rsZoneReport(&rsComm, &bbuf); ec < 0 ) {
+                const auto msg = fmt::format("[{}-{}] - Error obtaining zone report", __func__, __LINE__);
+                addRErrorMsg(&rsComm.rError, ec, msg.c_str());
+                THROW(ec, msg);
+            }
+
+            nlohmann::json zone_report = nlohmann::json::parse(
+                    std::string(static_cast<char *>(bbuf->buf), bbuf->len)
+            )["zones"];
```

Comment:
I'm pretty sure this can be rewritten as:
```cpp
const auto* buf = static_cast<char*>(bbuf->buf);
const auto zone_report = nlohmann::json::parse(buf, buf + bbuf->len).at("zones");
```

This avoids the construction of `std::string` which may result in a memory allocation. And if nothing modifies this object, then we can mark it as `const`.